runners/google-cloud-dataflow-java/pom.xml

```diff
@@ -299,7 +299,6 @@
     <dependency>
       <groupId>org.apache.beam</groupId>
       <artifactId>beam-runners-core-java</artifactId>
-      <scope>runtime</scope>
```

Comment:
Discussed offline. I believe we have good options for short term and long term to avoid bringing back this dependency:

**Longer term**: Put the functionality in `runners-core` but only invoke it in the Dataflow service. If this turns out to be easy then we should do it right away.

**Short-term**: Put the functionality elsewhere, in one of:

  - Dataflow runner module. This is my preference. Right now it is really a matter of how the Dataflow runner works that it is necessary to have this adapter. If another runner decides that it wants to go this same route, then we might be in the longer term scenario anyhow.
  - SDK. Mostly harmless & we should move it out prior to a stable release.
  - Some `io-core` module that is not quite the grab bag that `runners-core` is slated to be.

This highlights the issue that there are really two needs for a utility library:

1. Help implement the Beam model on the backend. This will generally impact a service which can be updated transparently and in an agile manner. It presupposes the service is aware of Beam constructs.
2. Help to put together a translation from a Beam pipeline to an underlying backend. This will generally occur in SDK-adjacent client-side code, which cannot be updated easily.

We can conflate the two without much of a downside, as long as we shade on both sides, but it helps me to think of them separately. The main risk being that we come to have too thick a client that is hard to update. Or we could split them pretty easily. I propose we wait and see.